.codesandbox/sandbox/src/end-to-end-test-helpers.ts

```diff
@@ -36,22 +36,41 @@ export function getHelpers(podRoot: string, session: Session) {
   }
 
   async function initialisePolicyResource() {
-    let inputRule = acp.createRule(policyResourceUrl + "#rule-public");
-    inputRule = acp.setPublic(inputRule, true);
+    let publicRule = acp.createRule(policyResourceUrl + "#rule-public");
+    publicRule = acp.setPublic(publicRule, true);
 
-    let inputPolicy = acp.createPolicy(
+    let publicReadPolicy = acp.createPolicy(
       policyResourceUrl + "#policy-publicRead"
     );
-    inputPolicy = acp.addRequiredRuleUrl(inputPolicy, inputRule);
-    inputPolicy = acp.setAllowModes(inputPolicy, {
+    publicReadPolicy = acp.addRequiredRuleUrl(publicReadPolicy, publicRule);
+    publicReadPolicy = acp.setAllowModes(publicReadPolicy, {
+      read: true,
+      append: false,
+      write: false,
+    });
+
+    let selfRule = acp.createRule(policyResourceUrl + "#rule-self");
+    selfRule = acp.addAgent(selfRule, session.info.webId!);
+    let selfWriteNoReadPolicy = acp.createPolicy(
+      policyResourceUrl + "#policy-selfWriteNoRead"
+    );
+    selfWriteNoReadPolicy = acp.addRequiredRuleUrl(selfWriteNoReadPolicy, selfRule);
+    selfWriteNoReadPolicy = acp.setAllowModes(selfWriteNoReadPolicy, {
+      read: false,
+      append: true,
+      write: true,
+    });
+    selfWriteNoReadPolicy = acp.setDenyModes(selfWriteNoReadPolicy, {
```

Comment:
Just pointing out, but I seem to be coming across this pattern more and more, i.e.:
```
  let x = createDataset();
  x = doActionOne(x, ...);
  x = doActionTwo(x, ...);
  x = doActionThree(x, ...);
```
...which really reads much cleaner as:
```
  const x = createDataset().doActionOne(...).doActionTwo(...).doActionThree(...);
```
but just pointing it out... :) !